src/test/CropJoin-unit.t.sol

```diff
@@ -560,4 +563,22 @@ contract CropUnitTest is TestBase {
         assertEq(adapter.stake(address(b)), 0);
         assertEq(bonus.balanceOf(address(b)), preBonusBal);
     }
+
+    function test_cage() public {
+        assertEq(adapter.live(), 1);
+        adapter.cage();
+        assertEq(adapter.live(), 0);
```

Comment:
I'd add logic either in this test or separate ones to confirm that `exit` and `flee` still work post-`cage`.